roles/kubernetes-apps/ingress_controller/cert_manager/templates/deploy-cert-manager.yml.j2

```diff
@@ -2,27 +2,32 @@
 apiVersion: apps/v1beta1
 kind: Deployment
 metadata:
-  name: cert-manager
+  name: cert-manager-{{ cert_manager_controller_image_tag }}
   namespace: {{ cert_manager_namespace }}
   labels:
     app: cert-manager
     chart: cert-manager-v0.4.1
     release: cert-manager
     heritage: Tiller
+    version: {{ cert_manager_controller_image_tag }}
 spec:
   replicas: 1
   selector:
     matchLabels:
       app: cert-manager
       release: cert-manager
+      version: {{ cert_manager_controller_image_tag }}
   template:
     metadata:
       labels:
         app: cert-manager
         release: cert-manager
+        version: {{ cert_manager_controller_image_tag }}
       annotations:
     spec:
+{% if rbac_enabled %}
```

Comment:
Rbac enabled or not, I think we can still create the service account regardless of the cluster policy.

Reply:
In tasks/main.yml, all service account related, e.g. sa, clusterrole, clusterrolebinding, etc are always create, as your suggestion ;-)

Reply:
so isn't safe to remove this condition and always attach the service account ?

Reply:
Seems this flag is for people who don’t hope to enable RBAC on pod, and I just clone this style of handling from elsewhere of kubespray’s addons.

For example, when implementing ingress-nginx for kubespray, the original manifests also come with two different example, with or without service account in pod, for with or without RBAC accordingly.

Reply:
@ant31 I check that most likely all of our `if rbac_enabled` use cases will always create service account, but optionally enable it in pod or not:

```
find roles -type f | xargs fgrep -nH 'if rbac_enabled' | sort
```
```
roles/dnsmasq/templates/dnsmasq-autoscaler.yml.j2:34:{% if rbac_enabled %}
roles/kubernetes-apps/ansible/templates/coredns-deployment.yml.j2:29:{% if rbac_enabled %}
roles/kubernetes-apps/ansible/templates/kubedns-autoscaler.yml.j2:67:{% if rbac_enabled %}
roles/kubernetes-apps/ansible/templates/kubedns-deploy.yml.j2:174:{% if rbac_enabled %}
roles/kubernetes-apps/ansible/templates/netchecker-server-deployment.yml.j2:36:{% if rbac_enabled %}
roles/kubernetes-apps/efk/elasticsearch/templates/elasticsearch-deployment.yml.j2:55:{% if rbac_enabled %}
roles/kubernetes-apps/efk/fluentd/templates/fluentd-ds.yml.j2:31:{% if rbac_enabled %}
roles/kubernetes-apps/efk/kibana/templates/kibana-deployment.yml.j2:49:{% if rbac_enabled %}
roles/kubernetes-apps/external_provisioner/cephfs_provisioner/templates/deploy-cephfs-provisioner.yml.j2:22:{% if rbac_enabled %}
roles/kubernetes-apps/helm/tasks/main.yml:34:    {% if rbac_enabled %} --service-account=tiller{% endif %}
roles/kubernetes-apps/ingress_controller/cert_manager/templates/deploy-cert-manager.yml.j2:28:{% if rbac_enabled %}
roles/kubernetes-apps/ingress_controller/ingress_nginx/templates/ds-ingress-nginx-controller.yml.j2:24:{% if rbac_enabled %}
roles/kubernetes-apps/policy_controller/calico/templates/calico-kube-controllers.yml.j2:24:{% if rbac_enabled %}
roles/kubernetes/master/templates/manifests/kube-controller-manager.manifest.j2:43:{% if rbac_enabled %}
roles/network_plugin/calico/templates/calico-node.yml.j2:25:{% if rbac_enabled %}
roles/network_plugin/canal/templates/canal-node.yaml.j2:22:{% if rbac_enabled %}
roles/network_plugin/cilium/templates/cilium-ds.yml.j2:37:{% if rbac_enabled %}
roles/network_plugin/contiv/templates/contiv-api-proxy.yml.j2:30:{% if rbac_enabled %}
roles/network_plugin/contiv/templates/contiv-netmaster.yml.j2:30:{% if rbac_enabled %}
roles/network_plugin/contiv/templates/contiv-netplugin.yml.j2:29:{% if rbac_enabled %}
roles/network_plugin/flannel/templates/cni-flannel.yml.j2:55:{% if rbac_enabled %}
```

Reply:
Using a service account does no harm.  Adding more and more conditions to templates make them harder to read, to maintain and to test the multiple paths. 
I would create a service account and bind it everytime regardless if rbac is enabled or not and then remove that conditions. 

Thanks for pointing other templates, I would prefer to fix all them rather than continue doing this.

Reply:
Which means we shall always create service account, and remove all rbac_enabled use cases in pod templates?

Agree with this idea too, but will this braking our legacy version support? If we just focus on recent 1~2 kubrnetes stable release support this shouldn’t be a problem.

Reply:
Why would it break kubernetes old support?
ServiceAccount has been there forever, the only thing that would eventually break is very old cluster doesn't know the 'RBAC' objects (clusterbinding, role...).
But rbac moved beta in 1.6, and officially only 4 versions are supported (https://kubernetes.io/docs/home/supported-doc-versions/) which is 1.7.

Reply:
@ant31 please review #3155?